src/rust/graph-query-service/tests/integration_test.rs

```diff
@@ -227,22 +279,75 @@ async fn test_query_single_node() -> Result<(), DynError> {
     assert_eq!(matched_graph.nodes.len(), 1);
     assert_eq!(matched_graph.edges.len(), 0);
 
-    let (returned_uid, returned_node) = matched_graph.nodes.into_iter().next().unwrap();
-    assert_eq!(returned_uid, uid);
-    assert_eq!(returned_uid, root_uid);
+    {
+        // Assertions on the root node
+        let (returned_uid, returned_node) = matched_graph.nodes.into_iter().next().unwrap();
+        assert_eq!(returned_uid, first_node_uid);
+        assert_eq!(returned_uid, root_uid);
 
-    assert_eq!(returned_node.node_type, node_type);
-    assert_eq!(returned_node.string_properties.prop_map.len(), 1);
+        assert_eq!(returned_node.node_type, process_node_type);
+        assert_eq!(returned_node.string_properties.prop_map.len(), 1);
 
-    let (returned_property_name, returned_property) = returned_node
-        .string_properties
-        .prop_map
-        .into_iter()
-        .next()
-        .unwrap();
+        let (returned_property_name, returned_property) = returned_node
+            .string_properties
+            .prop_map
+            .into_iter()
+            .next()
+            .unwrap();
+
+        assert_eq!(returned_property_name, "process_name".try_into()?);
+        assert_eq!(&returned_property, "chrome.exe");
+    }
+
+    // Now try to query the 'full' graph from that root uid
+    let graph_query = NodeQuery::root(process_node_type.clone())
+        .with_shared_edge(
+            forward_edge_name.clone(),
+            reverse_edge_name.clone(),
+            NodePropertyQuery::new(file_node_type.clone()),
+            // ??? what is init_edge for?
```

Comment:
@colin-grapl ok but what is it for tho

Reply:
A callback that allows you to modify the neighbor. You don't have a shared edge here so I'd suggest `with_edge_to`.

In both cases the API is designed to ensure that it is difficult (with_shared_edge) or impossible (with_edge_to) to construct an invalid query.